packages/switch/src/LionButtonSwitch.js

```diff
@@ -0,0 +1,95 @@
+import { LionButton } from '@lion/button';
+import { html, css } from '@lion/core';
+
+export class LionButtonSwitch extends LionButton {
+  static get properties() {
+    return {
+      readOnly: {
+        type: Boolean,
+        reflect: true,
+        attribute: 'readonly',
+      },
+      checked: {
+        type: Boolean,
+        reflect: true,
+      },
+    };
+  }
+
+  static get styles() {
+    return [
+      css`
+        :host {
+          display: inline-block;
+          position: relative;
+          width: 36px;
+          height: 16px;
+        }
+
+        .button-switch__track {
+          background: #eee;
+          width: 100%;
+          height: 100%;
+        }
+
+        .button-switch__thumb {
+          background: #ccc;
+          width: 50%;
+          height: 100%;
+          position: absolute;
+          top: 0;
+        }
+
+        :host([checked]) .button-switch__thumb {
+          right: 0;
+        }
+      `,
+    ];
+  }
+
+  render() {
+    return html`
+      <div class="button-switch__track"></div>
+      <div class="button-switch__thumb"></div>
+    `;
+  }
+
+  constructor() {
+    super();
+    this.type = 'button';
+    this.checked = false;
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    this._clickHandler = () => {
```

Comment:
Small remark: we usually do double underscores for handlers are not intended to be overwritten by Subclassers. Maybe a less generic name would also prevent chances of someone extending this class accidentally overriding it.

Reply:
updated, please confirm and "Resolve conversation"